jujugui/static/gui/src/app/jujulib/payment.js

```diff
@@ -237,22 +244,26 @@ var module = module;
       @public createPaymentMethod
       @param username {String} The user's username.
       @param token {String} A Stripe token.
+      @param methodName {String} A name for this payment method.
       @param callback {Function} A callback to handle errors or accept the
         data from the request. Must accept an error message or null as its
         first parameter and a payment method object as its second (see the
         getPaymentMethods docstring for the fields it returns).
     */
-    createPaymentMethod: function(username, token, callback) {
+    createPaymentMethod: function(username, token, methodName, callback) {
       const handler = (error, response) => {
         if (error !== null) {
           callback(error, null);
           return;
         }
         callback(null, this._parsePaymentMethod(response, true));
       };
-      const id = this._generatePaymentMethodName();
-      const url = `${this.url}/u/${username}/payment-methods/${id}`;
-      return jujulib._makeRequest(this.bakery, url, 'PUT', token, handler);
+      const url = `${this.url}/u/${username}/payment-methods`;
+      const payload = {
+        'payment-method-name': methodName,
```

Comment:
Am I reading right that from `payment-methods.js` will set this 'payment-method-name' to `null`? Is that ok to send in the payload?